lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,28 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+  #use a Hash to keep track of the number of duplicates
+  new_hash = Hash.new(0)
+  #iterate through each element in the array
+  arr.each do |idx|
```

Comment:
You're right, idx is a bit misleading. I will be more descriptive when naming my variables from now on.